src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -87,8 +87,11 @@ public void StartRequestProcessing()
             }
         }
 
-        public async void OnConnectionClosed()
+        public async void OnConnectionClosed(Exception ex)
         {
+            // Abort the connection (if it isn't already aborted)
+            _frame.Abort(ex);
```

Comment:
It replaces https://github.com/aspnet/KestrelHttpServer/blob/3b2d0a52f39fa024431c5b553f31bc4894291c0b/src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/OutputProducer.cs#L117